### ISSUE TITLE
1.0.4 breaks F# actors that send unit to actors that allow _ messages

### ISSUE THREAD

**Issue by raymens** [akkadotnet/akka.net#1285]
Actors that accept `_` in F# will not receive any messages send as a `()` *(unit)*. If the message is something else like a string, it does work. Or if the actor specifically accepts a `()` *(unit)* it also works.

Our codebase contains several pieces of code where this issue happens and as such we can't upgrade to 1.0.4 without altering our codebase.

Code sample that shows the issue:
```fsharp
open Akka.FSharp

let getWhateverHandler (mailbox : Actor<'a>) _ =
    printfn "Is this working?"

let system = System.create "my-system" (Configuration.load())
let parentActor = spawn system "Whatever" (actorOf2 getWhateverHandler)

// This doesn't work
parentActor <! ()

// This works
parentActor <! "This works"
```

**Comment by rogeralsing** [comment]
cc @Horusiath is this a result of the JObject hack?

**Comment by Aaronontheweb** [comment]
Time to add a unit test for this.

**Comment by rogeralsing** [comment]
Is this really a serialization bug? the example above does not use remoting and no serialization

**Comment by raymens** [comment]
Are these valid tests to use?

```
[<Fact>]
// FAILS currently
let ``actor that accepts _ will receive unit message`` () =    
    let timeoutConfig =
        """
        akka { 
            actor {
                ask-timeout = 5s
            }
        }
        """
        |> Configuration.parse 

    let getWhateverHandler (mailbox : Actor<_>) _ = 
        mailbox.Sender() <! "SomethingToReturn"

    let system = System.create "my-system" timeoutConfig
    let aref = spawn system "UnitActor" (actorOf2 getWhateverHandler)

    let response = aref <? () |> Async.RunSynchronously
    response
    |> equals "SomethingToReturn"

[<Fact>]
// SUCCEEDS
let ``actor that accepts unit will receive unit message`` () =    
    let timeoutConfig =
        """
        akka { 
            actor {
                ask-timeout = 5s
            }
        }
        """
        |> Configuration.parse 

    let getWhateverHandler (mailbox : Actor<unit>) () = 
        mailbox.Sender() <! "SomethingToReturn"

    let system = System.create "my-system" timeoutConfig
    let aref = spawn system "UnitActor" (actorOf2 getWhateverHandler)

    let response = aref <? () |> Async.RunSynchronously
    response
    |> equals "SomethingToReturn"
```

**Comment by rogeralsing** [comment]
@raymens looks good to me

**Comment by Aaronontheweb** [comment]
Can you send in a PR with that spec?

**Comment by raymens** [comment]
sure

**Comment by Danthar** [comment]
Closed: See #1326 for further discussion
